nmt.go

```diff
@@ -120,14 +143,30 @@ func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree {
 	}
 }
 
-// Prove leaf at index.
-// Note this is not really NMT specific but the tree supports inclusions proofs
+// Prove returns a NMT inclusion proof for the leaf at the supplied
+// index. Note this is not really NMT specific but the tree supports inclusions proofs
 // like any vanilla Merkle tree.
+// Prove is a thin wrapper around the ProveRange
 func (n *NamespacedMerkleTree) Prove(index int) (Proof, error) {
 	return n.ProveRange(index, index+1)
 }
 
-// ProveRange proves a leaf range [start, end].
+// ProveRange returns a Merkle inclusion proof for a specified range of leaves,
+// from start to end exclusive. The returned Proof structure contains the nodes
+// field, which holds the necessary tree nodes for the Merkle range proof in an
+// in-order traversal order. These nodes include the namespaced hash of the left
+// siblings for the proof of the leaf at index start, and the namespaced hash of
+// the right siblings for the proof of the leaf at index end.
+//
+// If the specified range [satrt,
+// end) exceeds the current range of leaves in the tree,
+// ProveRange returns an error together with an empty Proof with empty nodes and
+// start and end fields set to 0.
```

Comment:
the line wrapping here got messed up

```suggestion
// If the specified range [satrt, end) exceeds the current range of leaves in
// the tree, ProveRange returns an error together with an empty Proof with empty
// nodes and start and end fields set to 0.
```

Reply:
Fixed https://github.com/celestiaorg/nmt/commit/26b1ea676ac264bb2c6894efcd7a4f4aaf3924b3